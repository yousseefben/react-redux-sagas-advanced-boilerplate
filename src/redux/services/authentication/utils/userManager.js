import { createUserManager } from 'redux-oidc';

const userManagerConfig = {
  client_id: 'example-app',
  redirect_uri: `${window.location.protocol}//${window.location.hostname}${
    window.location.port ? `:${window.location.port}` : ''
  }/callback`,
  response_type: 'token id_token',
  // scope: 'openid profile https://www.googleapis.com/auth/youtube.readonly',
  authority: 'http://localhost:8080/auth/realms/test',
  silent_redirect_uri: `${window.location.protocol}//${
    window.location.hostname
  }${window.location.port ? `:${window.location.port}` : ''}/silent_renew.html`,
  automaticSilentRenew: true,
  filterProtocolClaims: true,
  loadUserInfo: true
};

const userManager = createUserManager(userManagerConfig);

export default userManager;
