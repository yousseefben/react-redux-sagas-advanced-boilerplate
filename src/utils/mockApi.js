export const mockSuccesfulResponse = (
  status = 200,
  method,
  returnBody,
  contentType
) => {
  global.fetch = jest.fn().mockImplementationOnce(
    () =>
      new Promise(resolve => {
        resolve({
          headers: {
            get: () => {
              return contentType || 'application/json';
            }
          },
          ok: true,
          status,
          json: () => returnBody || {},
          text: () => returnBody
        });
      })
  );
};

export const mockFailedResponse = error => {
  global.fetch = jest.fn().mockImplementationOnce(
    () =>
      new Promise(() => {
        throw new Error(error);
      })
  );
};
