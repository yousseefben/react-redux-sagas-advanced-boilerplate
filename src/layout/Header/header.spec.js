import React from 'react';

import { cleanup } from '@testing-library/react';
import { renderWithRedux } from '../../utils/test_utils';
import HeaderContent from './index';

afterEach(cleanup);

test('should render HeaderContent', async () => {
  const component = renderWithRedux(<HeaderContent />);

  expect(component.baseElement).toMatchSnapshot();
});
