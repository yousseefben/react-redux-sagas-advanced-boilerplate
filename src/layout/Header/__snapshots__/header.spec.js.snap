// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render HeaderContent 1`] = `
<body>
  <div>
    <ul
      class="ant-menu header ant-menu-light ant-menu-root ant-menu-horizontal"
      role="menu"
    />
  </div>
</body>
`;
